### PR TITLE
fix: changed error message when creating PP from SO

### PR DESCRIPTION
**Pull request by nmimsnikhil**
[Asana](https://app.asana.com/0/1192407899754728/1199581751419280/f)
changed error message when creating Production Plan from Sales Order using Action button.
image

### CODE REVIEW

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 1]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
...
+		items = []
 		for d in self.get('po_items'):
 			if not d.bom_no:
-				frappe.throw(_("Row {0}: Please select default BOM for Item {1} ".format(d.idx, d.item_code)))
+				items.append(d.item_code+": "+d.item_name)
+			if d.sales_order:
+				sales_order = d.sales_order
+		if sales_order:
+			items = list(dict.fromkeys(items))
+			frappe.throw(_("In Sales Order '{0}' Please set default BOM for Item {1}".format(sales_order,items)))
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]: Display an ordered list

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 2]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
 	def validate(self):
 		self.calculate_total_planned_qty()
 		self.set_status()
+		items = []
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py:21]:
```suggestion
		items = []
		show_default_bom_msg = False
```

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 3]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
...
 		for d in self.get('po_items'):
 			if not d.bom_no:
-				frappe.throw(_("Row {0}: Please select default BOM for Item {1} ".format(d.idx, d.item_code)))
+				items.append(d.item_code+": "+d.item_name)
+			if d.sales_order:
+				sales_order = d.sales_order
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]:
```suggestion
				show_default_bom_msg = True

```

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 4]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
...
+		if sales_order:
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]:
```suggestion
		if show_default_bom_msg:
```

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 5]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
...
+			items = list(dict.fromkeys(items))
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]: `items` is already a list

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 6]
```diff
@@ -18,9 +18,15 @@ class ProductionPlan(Document):
...
+			frappe.throw(_("In Sales Order '{0}' Please set default BOM for Item: <ol><li>{1}</li></ol>".format(sales_order,'<li>'.join(items))))
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]:
```suggestion
			frappe.throw(_("In Sales Order '{0}' Please set default BOM for Item: <ol>{1}</ol>".format(sales_order,'<li>'.join(items))))
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.py (hunk 6 above)
Comment (nmimsnikhil) [reply erpnext/manufacturing/doctype/production_plan/production_plan.py]: image

Comment (nmimsnikhil) [reply erpnext/manufacturing/doctype/production_plan/production_plan.py]: if i remove <li> after <ol>

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**